src/printer.js

```diff
@@ -1271,32 +1271,36 @@ function handleNewSetLiteral(path, print) {
 }
 
 function handleNewListInit(path, print) {
-  // Technically there are 2 ways to declare new list:
-  // - new String[]{};
-  // - new List<String>{};
-  // We use the 2nd format because it can handle any type of new list, e.g.
-  // This is correct:
-  // List<List<String>> data = new List<List<String>>{};
-  // But this is not:
-  // List<List<String>> data = new List<String>[]{};
-
-  // Other quirks:
-  // All of these are correct:
-  // Id[] ids = new Id[]{};
-  // Id[] ids = new Id[1];
-  // Id[] ids = new Id[anotherIds];
-  // But this is not correct:
-  // Id[] ids = new Id[1]{};
+  // We can delcare lists in the following ways:
+  // new Object[size];
+  // new Object[] { value, ... };
+  // new List<Object>(); // Provides AST consistency.
+  // new List<Object>(size);
+
+  // We use Object[size] if a literal number is provided.
+  // We use List<Object>(param) otherwise.
+  // This should provide compatibility for all known types without knowing
+  // if the parameter is a variable (copy constructor) or literal size.
+
   const expressionDoc = path.call(print, "expr", "value");
   const parts = [];
+  const typePart = path.map(print, "types");
```

Comment:
It looks like you are using this later to compare the values. Generally we only do `path.map(print, "blah")` if we intend to actually print it. If you need to get the value feel free to simply traverse the subtree:

```
const node = path.getValue();
// You can use `node.types` from here
```
We do it this way because calling `path.map(print)` adds some unnecessary stuffs to the `typePart` variable that is specific to printing.

Reply:
Thanks. I walked in to this project with exactly zero experience, so I didn't know I could. I'll try to fix this.